extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,57 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(function waitTillReady() {
+      // Wait until the image has completed.
+      if (!imgPlaceholder.complete) {
+        return timer.promise(10).then(waitTillReady);
```

Comment:
That `setInterval` thing could end up a nice addition to the timer library. E.g. `pollUntil(callback, maxTime)`.